### ISSUE TITLE
Disparity in `grpc-swift` version usage/resolution across `package.swift` files

### ISSUE THREAD

**Issue by ArielDemarco** [open-telemetry/opentelemetry-swift#661]
Hi there! Since the merge of [this pull request](https://github.com/open-telemetry/opentelemetry-swift/pull/459), I've noticed a discrepancy in the versioning of `grpc-swift`:
- In `Package.swift`, it uses `exact: 1.0.0`.
- In `Package@swift-5.6.swift` and `Package@swift-5.9.swift`, it uses `from: 1.0.0`.

While this generally doesn’t cause issues, it’s worth mentioning that there were some [tagging problems in grpc-swift version 1.24.0](https://github.com/grpc/grpc-swift/releases/tag/1.24.0), which can occasionally lead to package resolution issues.

To address this, I was wondering if we could unify the versioning approach. Some possible solutions that would also prevent the previously mentioned issue could include:

- Defining `exact: 1.0.0` across all `Package.swift` files.
- Defining `from: 1.24.1` across all `Package.swift` files.
- Defining `exact: 1.24.2` (the latest) across all `Package.swift` files.

If this sounds good, I’d be happy to open a PR and test the changes.

Let me know your thoughts!

**Comment by nachoBonafonte** [comment]
Yes, we should unify and  try to use the latest possible for each version.
